test/src/test/java/hudson/model/UpdateCenterErrorsTest.java

```diff
@@ -0,0 +1,185 @@
+        Page pageAfterClick = HtmlElementUtil.click(getCheckNow(p));
+        String page = pageAfterClick.getWebResponse().getContentAsString();
+
+        if(isSuccess)
+            Assert.assertFalse(page.contains(Messages.PluginManager_CheckUpdateServerError(message)));
+        else
+            Assert.assertTrue(page.contains(Messages.PluginManager_CheckUpdateServerError(message)));
+    }
+
+    @TestExtension("updateSiteReturn502Test")
+    public static final class FailingWith502UpdateCenterAction implements RootAction {
+
+        @Override
+        public @CheckForNull
+        String getIconFileName() {
+            return "gear2.png";
+        }
+
+        @Override
+        public @CheckForNull String getDisplayName() {
+            return "Update Site returning 502";
+        }
+
+        @Override
+        public String getUrlName() {
+            return "updateSite502";
+        }
+
+        public HttpResponse doGetJson(StaplerRequest request) {
+            return HttpResponses.error(502, "Gateway error");
+        }
+    }
+
+    @TestExtension("updateSiteWrongJsonTest")
+    public static final class FailingWithWrongJsonUpdateCenterAction implements RootAction {
+
+        @Override
+        public @CheckForNull
+        String getIconFileName() {
+            return "gear2.png";
+        }
+
+        @Override
+        public @CheckForNull String getDisplayName() {
+            return "Update Site returning wrong json";
+        }
+
+        @Override
+        public String getUrlName() {
+            return "updateSiteWrongJson";
+        }
+
+        public HttpResponse doGetJson(StaplerRequest request) {
+            HttpResponse r = new HttpResponse() {
+                @Override
+                public void generateResponse(StaplerRequest staplerRequest, StaplerResponse staplerResponse, Object o) throws IOException, ServletException {
+                    staplerResponse.setContentType("text/json");
+                    staplerResponse.setStatus(200);
+                    PrintWriter w = staplerResponse.getWriter();
+                    w.append("{wrongjson}");
+                }
+            };
+
+            return r;
+        }
+    }
+
+    @TestExtension("updateSiteRightJsonTest")
+    public static final class ReturnRightJsonUpdateCenterAction implements RootAction {
+
+        @Override
+        public @CheckForNull
+        String getIconFileName() {
+            return "gear2.png";
+        }
+
+        @Override
+        public @CheckForNull String getDisplayName() {
+            return "Update Site returning right json";
+        }
+
+        @Override
+        public String getUrlName() {
+            return "updateSiteRightJson";
+        }
+
+        // The url has to end in update-center.json. See: UpdateSite#getMetadataUrlForDownloadable
+        public void  doDynamic(StaplerRequest staplerRequest, StaplerResponse staplerResponse) throws ServletException, IOException {
```

Comment:
nit: (removal of a double spaces)

```suggestion
        public void doDynamic(StaplerRequest staplerRequest, StaplerResponse staplerResponse) throws ServletException, IOException {
```